src/pyodide/internal/snapshot.ts

```diff
@@ -162,17 +171,23 @@ export function getSnapshotSettings() {
   };
 }
 
+type DylinkInfo = {
+  [name: string]: { handles: string[] };
+} & {
+  settings?: { baselineSnapshot?: boolean };
+};
+
 /**
  * This records which dynamic libraries have open handles (handed out by dlopen,
  * not yet dlclosed). We'll need to track this information so that we don't
  * crash if we dlsym the handle after restoring from the snapshot
  */
-function recordDsoHandles(Module) {
-  const dylinkInfo = {};
+function recordDsoHandles(Module: Module) {
+  const dylinkInfo: DylinkInfo = {};
   for (const [handle, { name }] of Object.entries(
     Module.LDSO.loadedLibsByHandle,
   )) {
-    if (handle === 0) {
+    if (Number(handle) === 0) {
```

Comment:
Yeah I just noticed the other day that this would never be true...

Reply:
heh, that's why we're doing this :-)

Reply:
let's add a TODO so we don't forget :)

Reply:
Yeah I noticed while trying to upstream this stuff to Pyodide which has typescript.

Reply:
I don't think a TODO is necessary because the fix (converting to a number) is already implemented, right?